src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java

```diff
@@ -415,20 +420,42 @@ private void onResponseSignalCacheEntryAborted(
     private void onResponseSignalRequestGroupLeaderUpdated(
         SignalFW signal)
     {
-        final ArrayFW<HttpHeaderFW> requestHeaders = getRequestHeaders();
-        Consumer<ArrayFW.Builder<HttpHeaderFW.Builder, HttpHeaderFW>> mutator =
-            builder -> requestHeaders.forEach(h -> builder.item(item -> item.name(h.name()).value(h.value())));
-
-        connect = factory.writer.newHttpStream(requestGroup::newRequest, connectRouteId, connectInitialId,
-            factory.supplyTraceId.getAsLong(), mutator, this::onConnectMessage);
-
-        factory.writer.doHttpRequest(connect,
-                                     connectRouteId,
-                                     connectInitialId,
-                                     signal.traceId(),
-                                     authorization,
-                                     mutator);
-        cleanupRequestSlotIfNecessary();
+        if (requestGroup.isRequestGroupLeader(acceptReplyId))
+        {
+            if (requestSlot.value != NO_SLOT)
+            {
+                final ArrayFW<HttpHeaderFW> requestHeaders = getRequestHeaders();
+                Consumer<ArrayFW.Builder<HttpHeaderFW.Builder, HttpHeaderFW>> mutator =
+                    builder -> requestHeaders.forEach(h -> builder.item(item -> item.name(h.name()).value(h.value())));
+
+                connect = factory.writer.newHttpStream(requestGroup::newRequest,
+                                                       connectRouteId,
+                                                       connectInitialId,
+                                                       factory.supplyTraceId.getAsLong(),
+                                                       mutator,
+                                                       this::onConnectMessage);
+
+                factory.writer.doHttpRequest(connect,
+                                             connectRouteId,
+                                             connectInitialId,
+                                             signal.traceId(),
+                                             authorization,
+                                             mutator);
+                cleanupRequestSlotIfNecessary();
+            }
+            else
+            {
+                if (ifNoneMatch != null)
+                {
+                    send304();
+                }
```

Comment:
This block goes back and forth on visibility of `ifNoneMatch`, it assumes that `send304()` will use `ifNoneMatch` internally.

Suggest that either (a) `send304` is refactored to receive `ifNoneMatch` as only parameter, or (b) extract `ifNoneMatch != null` into a method called `canSend304()` instead.